Style the redirect form

MKodde opened the pull request:
## Background
This is the form EB pages use to send back SAML messages (supports both SAMLResponse and SAMLRequest) but I only know of it using it to send back SAMLResponses to the SP.

## Test notes
I added a functional testing endpoint to render the form. In order to prevent the infinite loop and test other stuff:

1. Manually remove the auto submit code from the body tag of the twig template to prevent a submit loop
2. You can test no-js capabilities by disabling JS in your browser (duh)

See: `https://engine.vm.openconext.org/functional-testing/send`

## Screenshots

Mobile:
![Screenshot from 2020-10-01 12-58-22](https://user-images.githubusercontent.com/28252948/94801103-d7ed9d80-03e5-11eb-8125-dd6c18e0a0fa.png)

Larger browser screen:
![Screenshot from 2020-10-01 12-50-25](https://user-images.githubusercontent.com/28252948/94800340-b9d36d80-03e4-11eb-8ae6-a20be14c87a7.png)

No-JS
![Screenshot from 2020-10-01 12-55-05](https://user-images.githubusercontent.com/28252948/94800772-5bf35580-03e5-11eb-9a2c-1d8f58f51e3e.png)

## Remarks
I'm aware the no-js styling is lacking in theming, and does not have good a11y feature (like outline on focus). For now I'd like to await a first review of the basic styling.

Badlapje commented on theme/base/stylesheets/pages/form.scss:
```diff
@@ -0,0 +1,22 @@
+.form {
+    > .form__content {
+        color: $loginBlue;
+        font-size: 16px;
```
$f-normal is beter.  Die gebruikt relatieve waarden in plaats van een absolute waarde en is dus beter voor zoom.

MKodde replied:
Thanks, got to get this into my system! I'm used to setting everything to an absolute value everywhere possible ;)

Badlapje commented on theme/base/stylesheets/pages/form.scss:
```diff
@@ -0,0 +1,22 @@
+    > .form__content {
+        color: $loginBlue;
+        font-size: 16px;
+        padding: 0 2rem calculateRem(40px);
```
niet zo belangrijk, maar je kan de calculateRem call vervangen door 2.5rem, en die van 20px door 1.25 rem.  Bespaart weer wat rekenkracht.

MKodde replied:
Thanks, I'm all for having a smaller footprint. If for convenience, maybe we should remove the calculate-rem function/mixin? Or is it usefull in some other way?

Badlapje commented on theme/base/stylesheets/pages/form.scss:
```diff
@@ -0,0 +1,22 @@
+
+        > picture.form__spinner {
+            > img {
+                padding: 2em 2em 0 0;
```
kies je hier bewust voor em?  Anders zou ik rem nemen.  Zie ook [deze link](https://zellwk.com/blog/rem-vs-em/): em leidt heel vaak tot verwarring omdat het in hetzelfde code-blok andere waarden kan aannemen + het kan onbedoeld veranderen als iemand de parent font size aanpast zonder te kijken down-stream.

MKodde replied:
Thanks for the link. I now see why using REM is usefull and more predictable overall. I used to use EM for everything without knowing the difference between the two :blush:

Badlapje commented on theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:
```diff
@@ -2,39 +2,51 @@
-        <meta content="ie=edge,chrome=1" http-equiv="x-ua-compatible">
-        <meta content="initial-scale=1.0,user-scalable=no,maximum-scale=1,width=device-width" name="viewport">
+        <meta content="IE=edge" http-equiv="x-ua-compatible">
+        <meta content="initial-scale=1.0,width=device-width" name="viewport">
```
🏅

Badlapje commented on theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:
```diff
@@ -2,39 +2,51 @@
+        </header>
 
-        <p>{{ 'processing_waiting'|trans }}</p>
+        <div class="form__content">
```
you could change this one out for a main element

MKodde replied:
The reason I refrained from doing this was that the parent main element is styled to have a white background with the rounded corners at the bottom. As this is a screen that flashes by for most users. I think it's nicer to have this themed less prominently. 

We should verify this with our PM's don't you think?

Badlapje replied:
Agreed.

Badlapje commented on theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:
```diff
@@ -2,39 +2,51 @@
+            <p>{{ 'processing_long'|trans }}</p>
 
-            {{ xtra|raw }}
+            <picture class="form__spinner">
```
if we don't use another image source, then using a plain img tag is better

MKodde replied:
Ah! Cool with me. Copy pasted this from another template. In the hope I was doing the right thing ;)

Badlapje commented on theme/base/templates/modules/Authentication/View/Proxy/form.html.twig:
```diff
@@ -2,39 +2,51 @@
-            <noscript><input type="submit" value="Submit" class="c-button"/></noscript>
-        </form>
+            <noscript>
+                <p>
```
you could move this to the noscript tag of the form itself, to save one noscript tag & put it all together (i'd have to run the html through a validator but 99% sure that's allowed)

MKodde replied:
:+1:

Badlapje commented on src/OpenConext/EngineBlockFunctionalTestingBundle/Resources/config/routing.yml:
```diff
@@ -55,6 +55,16 @@ functional_testing_consent:
     defaults:
         _controller: engineblock.functional_test.controller.consent:consentAction
 
+functional_testing_send_form:
```
🚀

Badlapje commented:
Not sure the blue font on blue background has enough contrast, i'll check that in 30

MKodde commented:
Thanks for the thorough review. I've addressed all of you feedback points. And in regards to:

1. Yes, changed it to the light grey background color
2. Ack
3. Thanks, have to get those mixins into my system!

See screenshot with where we're at at this point. I suggest we show this to the PM's as well. And align their suggestions in a successive PR

![Screenshot_2020-10-05 OpenConext - POST Data](https://user-images.githubusercontent.com/28252948/95050315-9d3c7b80-06eb-11eb-9b2a-98bde26c295e.png)

Badlapje commented:
Ziet er goed uit 💯
